v4/src/AddNodeModal.jsx

```diff
@@ -101,97 +194,12 @@ class AddNodeModal extends Component {
-
-                      {
-                      resourceType === 0 || resourceType === 1
-                        ? (
-                          <div>
-                            <Form.Control id="abc" name="abc" type="text" placeholder="Bitcoin is a p2p cash system" onChange={(ev) => this.setState({ resourceData: ev.target.value })} />
-                            <Form.Control.Feedback type="invalid">
-                            Please provide valid resource data
-                            </Form.Control.Feedback>
-                            <Form.Text>
-                            Resource data can be a description or hyperlink
-                            </Form.Text>
-                          </div>
-                        )
-                        : (
-                          <div>
-                            <Form.Control
-                              type="text"
-                              placeholder="Bitcoin whitepaper"
-                              onChange={(ev) => {
-                                const val = ev.target.value; // to save the virtual event
-                                this.setState((prevState) => ({
-                                  resourceData: {
-                                    ...prevState.resourceData,
-                                    text: val,
-                                  },
-                                }));
-                              }}
-                            />
-                            <Form.Control.Feedback type="invalid">
-                            Please provide valid resource link name
-                            </Form.Control.Feedback>
-                            <Form.Text>
-                            Resource link name
-                            </Form.Text>
-
-                            <Form.Control
-                              type="url"
-                              placeholder="https://bitcoin.org/bitcoin.pdf"
-                              onChange={(ev) => {
-                                const val = ev.target.value;
-                                this.setState((prevState) => ({
-                                  resourceData: {
-                                    ...prevState.resourceData,
-                                    link: val,
-                                  },
-                                }));
-                              }}
-                            />
-                            <Form.Control.Feedback type="invalid">
-                            Please provide valid resource link
-                            </Form.Control.Feedback>
-                            <Form.Text>
-                            Resource link URL
-                            </Form.Text>
-                          </div>
-                        )
-                    }
-
-
-                    </Form.Group>
-                  )
-
-              )
-              }
-
+            {this.pickedNode()}
```

Comment:
So awesome

Reply:
@rianadon @susan-lin "the real awesome lists were the friends we made along the way"

Reply:
omg LOL LL  @rustielin